From: Alban Browaeys <prahal@yahoo.com>
Date: Mon, 25 Apr 2011 15:55:43 +0200
Subject: [PATCH] Fix bus_manager_handler:

send manager "m" instead of null to bus_manager_append_n_jobs as it
expect the dbus manager as data nowadays.
Fix abort in systemd when calling this handler.
---
 src/dbus-manager.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/dbus-manager.c b/src/dbus-manager.c
index 4224afa..9776b0b 100644
--- a/src/dbus-manager.c
+++ b/src/dbus-manager.c
@@ -381,7 +381,7 @@ static DBusHandlerResult bus_manager_message_handler(DBusConnection *connection,
                 { "org.freedesktop.systemd1.Manager", "LogLevel",      bus_manager_append_log_level,  "s",  m, bus_manager_set_log_level },
                 { "org.freedesktop.systemd1.Manager", "LogTarget",     bus_manager_append_log_target, "s",  m, bus_manager_set_log_target },
                 { "org.freedesktop.systemd1.Manager", "NNames",        bus_manager_append_n_names,    "u",  m                  },
-                { "org.freedesktop.systemd1.Manager", "NJobs",         bus_manager_append_n_jobs,     "u",  NULL               },
+                { "org.freedesktop.systemd1.Manager", "NJobs",         bus_manager_append_n_jobs,     "u",  m                  },
                 { "org.freedesktop.systemd1.Manager", "NInstalledJobs",bus_property_append_uint32,    "u",  &m->n_installed_jobs },
                 { "org.freedesktop.systemd1.Manager", "NFailedJobs",   bus_property_append_uint32,    "u",  &m->n_failed_jobs  },
                 { "org.freedesktop.systemd1.Manager", "Progress",      bus_manager_append_progress,   "d",  m                  },
-- 
1.7.5.rc3
